src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AFModel.java

```diff
@@ -0,0 +1,239 @@
+        final double logNumerator = hetLogLikelihood(parameters.withIncrementedAlpha(k), count, indicator) + Gamma.logGamma(alpha + k) - (alpha + k)*log(beta);
+        final double logDenominator = hetLogLikelihood(parameters, count, indicator) + Gamma.logGamma(alpha) - alpha*log(beta);
+        return Math.exp(logNumerator - logDenominator);
+    }
+
+    /**
+     * Adds {@code numSamples - numBurnIn} Markov-Chain Monte-Carlo samples of the parameter posteriors (generated using
+     * Gibbs sampling) to the collections held internally.  The current {@link AFState} held internally is used
+     * to initialize the Markov Chain.
+     * @param numSamples    total number of samples per posterior
+     * @param numBurnIn     number of burn-in samples to discard
+     */
+    public void fitMCMC(final int numSamples, final int numBurnIn) {
+        //run MCMC
+        final GibbsSampler<AFState, AFData> gibbsSampler = new GibbsSampler<>(numSamples + numBurnIn, model);
+        gibbsSampler.runMCMC();
+
+        //update posterior samples
+        meanBiasSamples.addAll(gibbsSampler.getSamples(AFState.MEAN_BIAS_NAME, Double.class, numBurnIn));
+        biasRateSamples.addAll(gibbsSampler.getSamples(AFState.BIAS_RATE_NAME, Double.class, numBurnIn));
+        outlierProbabilitySamples.addAll(gibbsSampler.getSamples(AFState.P_OUTLIER_NAME, Double.class, numBurnIn));
+        minorFractionsSamples.addAll(gibbsSampler.getSamples(AFState.MINOR_FRACTIONS_NAME, AFState.MinorFractions.class, numBurnIn));
+    }
+
+
+    public List<Double> getmeanBiasSamples() {
+        return Collections.unmodifiableList(meanBiasSamples);
+    }
+    public List<Double> getBiasRateSamples() {
+        return Collections.unmodifiableList(biasRateSamples);
+    }
+    public List<Double> getOutlierProbabilitySamples() { return Collections.unmodifiableList(outlierProbabilitySamples); }
+    public List<AFState.MinorFractions> getMinorFractionsSamples() { return Collections.unmodifiableList(minorFractionsSamples); }
+
+    public List<List<Double>> getMinorFractionSamplesBySegment() {
+        final List<List<Double>> result = new ArrayList<>();
+        for (int segment = 0; segment < numSegments; segment++) {
+            final List<Double> thisSegment = new ArrayList<>();
+            for (final AFState.MinorFractions sample : minorFractionsSamples) {
+                thisSegment.add(sample.get(segment));
+            }
+            result.add(thisSegment);
+        }
+        return result;
+    }
+
+    //compute log(e^a + e^b + e^c) = log(e^M [e^(a-M) + e^(b-M) + e^(c-M)]) where M = max(a,b,c)
+    private static double logSumLog(final double a, final double b, final double c) {
+        final double M = max(a, b, c);
+        return M + Math.log(Math.exp(a-M) + Math.exp(b-M) + Math.exp(c-M));
+    }
+
+    //TODO: move to Utils (also it's redundant with something in AFInitializer)
+    private static double max(final double a, final double b, final double c) {
```

Comment:
Use ``import com.google.common.primitives.Doubles``  This is already in hellbender...

``Doubles.max(a,b,c);``